File: packages/javascript/lib/math/numPick.ts
export function numPick(min: number, max: number): number {
	if (!min && !max) {
		return Math.random() > 0.5 ? 1 : 0;
	}

	const limit = !max ? min : max;
	const offset = !max || min >= max ? null : min;

	return Math.floor(Math.random() * (offset ? limit - offset + 1 : limit + 1)) + (offset || 0);
}

File: packages/python/qsu/math/numPick.py
import random


def numPick(min: int, max: int) -> int:
	if not min and not max:
		return 1 if random.random() > 0.5 else 0

	limit = min if not max else max
	offset = None if (not max or min >= max) else min

	return int(random.random() * ((limit - offset + 1) if offset else (limit + 1))) + (
		offset or 0
	)
